runtimes/native/src/runtime.c

```diff
@@ -128,7 +135,12 @@ void w4_runtimeTextUtf16 (const uint16_t* str, int byteLength, int x, int y) {
 
 void w4_runtimeTone (int frequency, int duration, int volume, int flags) {
     // printf("tone: %d, %d, %d, %d\n", frequency, duration, volume, flags);
-    w4_apuTone(frequency, duration, volume, flags);
+    int channelIdx = flags & 0x3;
+    bufferedToneCalls[channelIdx].active = true;
+    bufferedToneCalls[channelIdx].frequency = frequency;
+    bufferedToneCalls[channelIdx].duration = duration;
+    bufferedToneCalls[channelIdx].volume = volume;
+    bufferedToneCalls[channelIdx].flags = flags;
```

Comment:
Not sure if keeping APU state outside of the APU is desired here.

The APU already has internal global state, so maybe this logic could be placed in `apu.c` instead to keep it isolated?